metabase/metabase
# Refactor action entity create and update extensions

Reorganizes query action update logic to make it simpler to update actions from anywhere except the action editor. This is needed to unblock action archiving as we're doing it via the `PUT /api/action/:id { archived: true }` endpoint.

**Background**

Before we needed to archive actions, the only place from where we triggered action updates was the `ActionCreator`. It uses `NativeQueryEditor` that was built for questions, so it expects a `question` prop and works with its `query`. Refactoring `NativeQueryEditor` is a complex and, most likely, a long task, so we used a workaround.

Query actions have a similar structure to questions (both have `name`, `description`, `visualization_settings` and identical `dataset_query`), so we create a semi-valid `Question` instance from an action and pass it to the `NativeQueryEditor`. We made the `Actions` entity `api.create` and `api.update` extensions accept a `question` parameter. They'd pull out all the necessary info from the `question` instance and then would call an action API endpoint.

**Problem and solution**

Now when we want to let people archive actions, the required `question` parameter became an issue as it doesn't make any sense in this use-case. 

This PR moves all the `question` related logic inside the `ActionCreator`. This workaround is introduced exclusively for the `NativeQueryEditor`, so it makes much more sense to keep it inside as an implementation detail. It also moves the `CreateActionForm` component from `actions/containers` inside the `ActionCreator`. We also use the `question` workaround in there, so it made it possible to keep this logic isolated inside the editor component. Also, the modal itself doesn't seem to be an indepent component that can be used without the `ActionCreator`

### To Verify

1. Open the model detail, click the "Actions" tab
2. Play around with creating and editing query actions, ensure everything works the same as on `master`
